src/backends/torch/torchdataaug.cc

```diff
@@ -79,69 +159,133 @@ namespace dd
       {
         cv::Mat dst;
         cv::transpose(src, dst);
-        cv::flip(dst, src, 0);
+        cv::flip(dst, src, 1);
       }
+    return rot;
   }
 
-  void TorchImgRandAugCV::applyCrop(cv::Mat &src)
+  void
+  TorchImgRandAugCV::applyRotateBBox(std::vector<std::vector<float>> &bboxes,
+                                     const float &img_width,
+                                     const float &img_height, const int &rot)
   {
+    std::vector<std::vector<float>> nbboxes;
+    for (size_t i = 0; i < bboxes.size(); ++i)
+      {
+        std::vector<float> bbox = bboxes.at(i);
+        std::vector<float> nbox;
+        if (rot == 1) // 90
+          {
+            nbox.push_back(bbox[1]);              // xmin <- ymin
+            nbox.push_back(img_height - bbox[2]); // ymin <- height-xmax
```

Comment:
`width-xmax` rather than `height-xmax`?

Reply:
I don't think so, because at this stage, the input image has already been rotated.

Reply:
If img_height is the image height after rotation, then the `rot == 3` block is wrong, since it uses `height` as image height before rotation.

Reply:
The code is correct, what has changed is https://github.com/jolibrain/deepdetect/pull/1276/files#diff-d9fef5075d6397cff9fb5b52d7a57b88f9c388c96f51907d8bde5a5d0f1a1c10R161 and the comment should be that's 90, and the comment that says 90 should say 170, as flipping with 0 means counterclockwise.

Since OpenCV 3.2, there's a `cv::rotate` function, maybe I could just use it, would make `applyRotate` simpler.

Reply:
https://stackoverflow.com/questions/2259678/easiest-way-to-rotate-by-90-degrees-an-image-using-opencv

Reply:
@Bycob you are right, there was a bug visible when using non-squared images, this is now fixed!